File: launchdarkly/csharp/src/OSEG.LaunchDarklyExamples/GetPullRequestsExample.cs
using System;
using System.Collections.Generic;
using System.IO;
using System.Text.Json;

using Org.LaunchDarklyTools.Api;
using Org.LaunchDarklyTools.Client;
using Org.LaunchDarklyTools.Model;

namespace OSEG.LaunchDarklyExamples;

public class GetPullRequestsExample
{
    public static void Run()
    {
        var config = new Configuration();
        config.ApiKey.Add("Authorization", "YOUR_API_KEY");

        try
        {
            var response = new InsightsPullRequestsBetaApi(config).GetPullRequests(
                projectKey: "projectKey_string",
                environmentKey: null,
                applicationKey: null,
                status: null,
                query: null,
                limit: null,
                expand: null,
                sort: null,
                from: DateTime.Parse("None"),
                to: DateTime.Parse("None"),
                after: null,
                before: null
            );

            Console.WriteLine(response);
        }
        catch (ApiException e)
        {
            Console.WriteLine("Exception when calling InsightsPullRequestsBetaApi#GetPullRequests: " + e.Message);
            Console.WriteLine("Status Code: " + e.ErrorCode);
            Console.WriteLine(e.StackTrace);
        }
    }
}

File: launchdarkly/typescript-axios/src/GetPullRequestsExample.ts
import axios, { AxiosError } from "axios";
import * as api from "launchdarkly_client"

const configuration = new api.Configuration({
  apiKey: "YOUR_API_KEY",
});

new api.InsightsPullRequestsBetaApi(configuration).getPullRequests(
  "projectKey_string", // projectKey
  undefined, // environmentKey
  undefined, // applicationKey
  undefined, // status
  undefined, // query
  undefined, // limit
  undefined, // expand
  undefined, // sort
  undefined, // from
  undefined, // to
  undefined, // after
  undefined, // before
).then(response => {
  console.log(response.data);
}).catch((error: Error | AxiosError) => {
  console.log("Exception when calling InsightsPullRequestsBetaApi#getPullRequests:");
  axios.isAxiosError(error) ? console.log(error.message) : console.log(error);
});
